File: app/src/main/java/com/kevin/kggank/ui/presenter/ListPresenter.java
package com.kevin.kggank.ui.presenter;

import com.kevin.kggank.base.BasePresenter;
import com.kevin.kggank.entity.GanhuoListEntity;
import com.kevin.kggank.ui.model.ListModel;
import com.kevin.kggank.ui.view.IlistView;

import io.reactivex.subscribers.DisposableSubscriber;

/**
 * Created by kevin on 16/5/18.
 */
public class ListPresenter extends BasePresenter<IlistView> {

    private ListModel model = new ListModel(this);

    public ListPresenter(IlistView mView) {
        super(mView);

        getGirlInfo(1);
    }

    public void getGirlInfo(int pageNo) {
        addSubscribe(model.getGirlInfo(pageNo)
                .subscribeWith(new DisposableSubscriber<GanhuoListEntity>() {
                    @Override
                    public void onNext(GanhuoListEntity ganhuoListEntity) {
                        if (ganhuoListEntity == null ||
                                ganhuoListEntity.getResults() == null
                                || ganhuoListEntity.getResults().size() == 0)
                            return;

                        getView().getGanhuoListGril(ganhuoListEntity.getResults());
                    }

                    @Override
                    public void onError(Throwable t) {

                    }

                    @Override
                    public void onComplete() {

                    }
                }));
    }
}

File: app/src/main/java/com/kevin/kggank/ui/presenter/ListPresenter.kt
package com.kevin.kggank.ui.presenter

import com.kevin.kggank.entity.GanhuoListEntity
import com.kevin.kggank.ui.model.ListModel
import com.kevin.kggank.ui.view.IlistView
import com.kevin.kglib.base.LibBasePresenter
import io.reactivex.subscribers.DisposableSubscriber

/**
 * Created by kevin on 16/5/18.
 */
class ListPresenter
constructor(mView: IlistView) : LibBasePresenter<IlistView>(mView) {

    private val model = ListModel(this)

    init {
        getGirlInfo(1)
    }

    fun getGirlInfo(pageNo: Int) {
        addSubscribe(model.getGirlInfo(pageNo)
                .subscribeWith(object : DisposableSubscriber<GanhuoListEntity>() {
                    override fun onNext(ganhuoListEntity: GanhuoListEntity?) {
                        if (ganhuoListEntity == null ||
                                ganhuoListEntity.results == null
                                || ganhuoListEntity.results.size == 0)
                            return

                        getView().getGanhuoListGril(ganhuoListEntity.results)
                    }

                    override fun onError(t: Throwable) {

                    }

                    override fun onComplete() {

                    }
                }))
    }
}
